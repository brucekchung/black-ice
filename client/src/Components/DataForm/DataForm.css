.DataForm input {
  min-width: 175px;
  width: 22%;
}

.data-form-set {
  display: flex;
  justify-content: space-between;
  margin: auto;
  margin-bottom: 20px;
  width: 85%;
}

.DataForm button {
  margin: 10px;
  width: 115px;
}

.add-remove-buttons {
  margin: auto;
  text-align: right;
  width: 85%;
}

.add-remove-buttons button {
  margin: 0;
  margin-left: 20px;
}

@media screen and (max-width: 1020px) {
  .data-form-set {
    display: block;
    width: 66%;
  }

  .data-form-set input {
    display: block;
    margin: 20px 0;
    width: 100%;
  }

  .add-remove-buttons {
    width: 70%;
  }
}