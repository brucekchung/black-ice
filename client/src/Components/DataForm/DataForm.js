import React, { Component } from 'react';

class DataForm extends Component {
  renderInputs = () => {
    return this.props.data.map((item, index) => {
      return (
        <div key={ index } className={ index }>
          <h4>Data Point { index + 1 }</h4>
          <input type="text"  
                 name="name"
                 className="name"
                 placeholder="Name"
                 onChange={ this.props.handleChange }
                 value={ item.name } />
          <input type="text"
                 name="date_collected"
                 placeholder="Date collected"
                 onChange={ this.props.handleChange }
                 value={ item.date_collected } />
          <input type="text"
                 name="reflectance"
                 placeholder="Reflectance"
                 onChange={ this.props.handleChange }
                 value={ item.reflectance } />
          <input type="text"
                 name="wavelength"
                 placeholder="Wavelength"
                 onChange={ this.props.handleChange }
                 value={ item.wavelength } />
        </div>
      )
    })
  }

  render() {
    return (
      <div>
        { this.renderInputs() }
        <button onClick={ this.props.handleClick }>Add Data</button>
        <button onClick={ this.props.handleClick }>Remove</button>
      </div>
    )
  }
}

export default DataForm
