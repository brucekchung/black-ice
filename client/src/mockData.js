export const allData = [{
    sample_id: '1',
    name: 'Base Camp',
    date_collected: '08/19/18',
    reflectance: '89',
    wavelength: '20',
    location_id: '1',
    altitude: '4000m',
    lat: '1',
    lng: '-1',
    region: 'Ishinca',
    country: 'Peru'
  }, {
    sample_id: '2',
    name: 'Camp 1',
    date_collected: '08/20/18',
    reflectance: '70',
    wavelength: '110',
    location_id: '2',
    altitude: '5000m',
    lat: '2',
    lng: '-2',
    region: 'Pisco',
    country: 'Peru'       
}]

export const locationsOnly = [{
    id: '1',
    name: 'Base Camp',
    lat: '1',
    lng: '-1',
    alt: '4000m',
    region: 'Ishinca',
    country: 'Argentina'    
  }, {
    id: '2',
    name: 'Camp 1',
    lat: '2',
    lng: '-2',
    alt: '5000m',
    region: 'Pisco',
    country: 'Peru'
  }, {
    id: '3',
    name: 'Camp 2',
    lat: '1',
    lng: '-2',
    alt: '6000m',
    region: 'Ishinca',
    country: 'Argentina'
}]

export const dataOnly = [{
    id: '1',
    name: 'WP 1',
    date_collected: '08/19/18',
    reflectance: '89',
    wavelength: '20',
    location_id: '1'
  }, {
    id: '2',
    name: 'WP 22',
    date_collected: '08/20/18',
    reflectance: '70',
    wavelength: '110',
    location_id: '2'
}]