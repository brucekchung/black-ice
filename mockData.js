export const allData = [{
    id: '1',
    name: 'Base Camp',
    date_collected: '08/19/18',
    reflectance: '89%',
    wavelength: '20um',
    altitude: '4000m',
    latlng: '1, -1',
    region: 'Ishinca'
    country: 'test'
  }, {
    id: '2',
    name: 'Camp 1',
    date_collected: '08/20/18',
    reflectance: '70%',
    wavelength: '110um',
    altitude: '5000m',
    latlng: '2, -2',
    region: 'Pisco'
    country: 'test2'       
}]

export const allLocations = [{
    id: '1',
    name: 'Base Camp',
    lat: '1',
    lng: '-1',
    altitude: '4000m',
    region: 'Ishinca',
    country: 'Peru'    
  }, {
    id: '2',
    name: 'Camp 1',
    lat: '2',
    lng: '-2',
    altitude: '5000m',
    region: 'Pisco',
    country: 'Peru'
}]

export const data = [{
    id: '1',
    name: 'WP 1',
    date_collected: '08/19/18',
    reflectance: '89%',
    wavelength: '20um',
    location_id: '1'
  }, {
    id: '2',
    name: 'WP 22',
    date_collected: '08/20/18',
    reflectance: '70%',
    wavelength: '110um',
    location_id: '2'
}]