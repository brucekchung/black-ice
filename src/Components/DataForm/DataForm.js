import React, { Component } from 'react';

class DataForm extends Component {
  constructor(props) {
    super(props)

    this.state = {
      repeatedItems: [{
          name: '',
          date_collected: '',
          reflectance: '',
          wavelength: ''
      }]
    }
  }

  renderInputs = () => {
    return this.state.repeatedItems.map((item, index) => {
      const state = this.state.repeatedItems[index]
      return (
        <div key={ index }>
          <h4>Data Point { index + 1 }</h4>
          <input type="text"  
                 name="name"
                 placeholder="Name"
                 value={ state.name } />
          <input type="text"
                 name="date_collected"
                 placeholder="Date collected"
                 value={ state.date_collected } />
          <input type="text"
                 name="reflectance"
                 placeholder="Reflectance"
                 value={ state.reflectance } />
          <input type="text"
                 name="wavelength"
                 placeholder="Wavelength"
                 value={ state.wavelength } />
        </div>
      )
    })
  }

  addData = () => {
    const newSet = {
      name: '',
      date_collected: '',
      reflectance: '',
      wavelength: ''
    }
    const newItemsArray = [...this.state.repeatedItems, newSet]

    this.setState({ repeatedItems: newItemsArray })
  }

  removeData = () => {
    if (this.state.repeatedItems.length > 1) {
      const allItems = [...this.state.repeatedItems]

      allItems.pop();
      this.setState({ repeatedItems: allItems })
    }
  }

  handleClick = (e) => {
    e.preventDefault()
    e.target.innerText === 'Add Data' ? this.addData() : this.removeData()
  }

  render() {
    return (
      <div>
        { this.renderInputs() }
        <button onClick={ this.handleClick }>Add Data</button>
        <button onClick={ this.handleClick }>Remove</button>
      </div>
    )
  }
}

export default DataForm