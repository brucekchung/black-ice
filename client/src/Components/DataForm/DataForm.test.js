import React from 'react'
import DataForm from './DataForm'
import { shallow } from 'enzyme'
import { dataOnly } from '../../mockData'

describe('DataForm', () => {
  let wrapper;

  beforeEach(() => {
    const wrapper = shallow(<DataForm data={ dataOnly }/>, { disableLifecycleMethods: true })
  })

  it('should match snapshot upon load', () => {
    expect(wrapper).toMatchSnapshot()
  })

})