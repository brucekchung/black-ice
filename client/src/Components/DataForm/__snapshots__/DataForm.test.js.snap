// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataForm should match snapshot upon load 1`] = `undefined`;
